rust/macros/of_id.rs

```diff
@@ -0,0 +1,68 @@
+// SPDX-License-Identifier: GPL-2.0
+
+use proc_macro::{TokenStream, TokenTree};
+
+pub(crate) fn derive_get_id_info(ts: TokenStream) -> TokenStream {
+    let tokens: Vec<_> = ts.into_iter().collect();
+    let mut type_name = String::from("Struct");
+
+    tokens
+        .iter()
+        .find_map(|token| match token {
+            TokenTree::Ident(ident) => match ident.to_string().as_str() {
+                "struct" => Some(1),
+                _ => None,
+            },
+            _ => None,
+        })
+        .expect("#[GetIdInfo] should be applied to a struct");
+
+    let mut content = tokens.into_iter();
+    while let Some(token) = content.next() {
+        match token {
+            TokenTree::Ident(ident) if ident.to_string() == "struct" => {
+                if let Some(TokenTree::Ident(ident)) = content.next() {
+                    type_name = ident.to_string()
+                }
+            }
+            _ => (),
+        }
+    }
+
+    println!("value {:?}", type_name);
+
+    format!(
+        "
+            impl<T: Driver> {name}<T> {{
```

Comment:
`#[derive(Foo)]` generally implements a trait called `Foo` for the respective type. Not sure if we should introduce a trait `GetIdInfo`, or not.